Source/WebKit/UIProcess/Cocoa/WebProcessPoolCocoa.mm

```diff
@@ -645,6 +645,58 @@ static void logProcessPoolState(const WebProcessPool& pool)
 {
     m_weakObserver = adoptNS([[WKProcessPoolWeakObserver alloc] initWithWeakPtr:*this]);
 
+#if ENABLE(NOTIFYD_BLOCKING_IN_WEBCONTENT)
+    constexpr ASCIILiteral notificationMessages[] = {
+        "com.apple.WebKit.LibraryPathDiagnostics"_s,
+        "com.apple.WebKit.dumpUntrackedMallocs"_s,
+        "org.WebKit.memoryWarning"_s,
+        "org.WebKit.memoryWarning.begin"_s,
+        "org.WebKit.memoryWarning.end"_s,
+        "org.WebKit.lowMemory"_s,
+        "org.WebKit.lowMemory.begin"_s,
+        "org.WebKit.lowMemory.end"_s,
+        "com.apple.WebKit.fullGC"_s,
+        "com.apple.WebKit.deleteAllCode"_s,
+        "com.apple.WebKit.dumpGCHeap"_s,
+        "com.apple.WebKit.showBackForwardCache"_s,
+        "com.apple.WebKit.showMemoryCache"_s,
+        "com.apple.WebKit.showRenderTree"_s,
+        "com.apple.WebKit.showLayerTree"_s,
+        "com.apple.WebKit.showGraphicsLayerTree"_s,
+        "com.apple.WebKit.showPaintOrderTree"_s,
+        "com.apple.WebKit.showLayoutTree"_s,
+        "com.apple.WebKit.showAllDocuments"_s,
+        "com.apple.WebKit.logMemStats"_s,
+        "com.apple.WebKit.logPageState"_s
+    };
+    m_notifyTokens = WTF::map(notificationMessages, [this](const ASCIILiteral& message) -> int {
+        int notifyToken;
+        notify_register_dispatch(message, &notifyToken, dispatch_get_main_queue(), ^(int token) {
+            if (!m_processes.isEmpty()) {
+                String messageString(message);
+                for (auto& process : m_processes)
+                    process->send(Messages::WebProcess::PostNotification(messageString), 0);
+            }
+        });
+        return notifyToken;
+    });
+
+    NSString* nsNotificationMessages[] = {
+        NSProcessInfoPowerStateDidChangeNotification
+    };
+    m_notificationObservers.reserveInitialCapacity(std::size(nsNotificationMessages));
```

Comment:
Again, we should try to use the `WTF::compactMap` formulation here:

```
m_notificationObservers = WTF::compactMap(nsNotificationMessages, [this](NSString* message) -> RetainPtr<NSObject>  {
            RetainPtr<NSObject> observer = [[NSNotificationCenter defaultCenter] addObserverForName:message object:nil queue:[NSOperationQueue currentQueue] usingBlock:^(NSNotification *notification) {
            if (!m_processes.isEmpty()) {
                String messageString(message);
                for (auto& process : m_processes)
                    process->send(Messages::WebProcess::PostObserverNotification(message), 0);
            }
        }];
        return observer;
});
```

Reply:
`compactMap` should return a Vector of all non-nil RetainPtrs.

Reply:
Nice, thanks! I will address this in the upcoming patch :)